packages/contracts-bedrock/test/scripts/L2Genesis.t.sol

```diff
@@ -0,0 +1,232 @@
+        assertNotEq(keccak256(_alloc.code), keccak256(bytes("0x")));
+
+        if (_hasStorageSlotsSet(_alloc.addr)) {
+            assertNotEq(_alloc.storageData.length, 0);
+        } else {
+            assertEq(_alloc.storageData.length, 0);
+        }
+    }
+
+    function _checkDevAccount(Alloc memory _alloc) internal {
+        if (!cfg.fundDevAccounts()) {
+            revert(
+                string.concat(
+                    "Dev account ",
+                    vm.toString(_alloc.addr),
+                    " present in allocs, but fundDevAccounts is not enabled in deploy config"
+                )
+            );
+        }
+
+        assertEq(vm.parseUint(vm.toString(_alloc.balance)), DEV_ACCOUNT_FUND_AMT);
+        assertEq(_alloc.code, hex"");
+        assertEq(_alloc.nonce, hex"00");
+        assertEq(_alloc.storageData.length, 0);
+    }
+
+    /// @notice Parses a given `_filePath` into a `Alloc[]`.
+    function _parseAllocs(string memory _filePath) internal returns (Alloc[] memory) {
+        string[] memory cmd = new string[](3);
+        cmd[0] = "bash";
+        cmd[1] = "-c";
+        cmd[2] = string.concat(
+            Executables.jq,
+            " -cr 'to_entries | map({addr: .key, balance: .value.balance, code: .value.code, nonce: .value.nonce, storageData: (.value.storage | to_entries | map({key: .key, value: .value}))})' ",
+            _filePath
+        );
+        bytes memory result = vm.ffi(cmd);
+        bytes memory parsedJson = vm.parseJson(string(result));
+        return abi.decode(parsedJson, (Alloc[]));
+    }
+
+    /// @notice Returns whether the given address has a given prefix.
+    function _addressHasPrefix(address _addr, uint160 _prefix, uint160 _mask) internal pure returns (bool) {
+        uint160 numericAddress = uint160(_addr);
+        return (numericAddress & _mask) == _prefix;
+    }
+
+    /// @notice Returns whether a given address has the expected predeploy proxy prefix.
+    function _isProxyAddress(address _addr) internal pure returns (bool) {
+        return _addressHasPrefix(
+            _addr,
+            uint160(0x4200000000000000000000000000000000000000),
+            uint160(0xfFFFFfffFFFfFfFFffFFFfFfffFFfFfF00000000)
+        );
+    }
+
+    /// @notice Returns whether a given address has the expected predeploy implementation prefix.
+    function _isImplementationAddress(address _addr) internal pure returns (bool) {
+        return _addressHasPrefix(
+            _addr,
+            uint160(0xc0D3C0d3C0d3C0D3c0d3C0d3c0D3C0d3c0d30000),
+            uint160(0xfFfffFFFfffFFfFFFFffFFFFffffFfFFFFff0000)
+        );
+    }
+
+    function _isDevAccount(address _addr) internal pure returns (bool) {
+        for (uint256 i; i < cfg.getDevAccounts().length; i++) {
+            if (_addr == cfg.getDevAccounts()[i]) return true;
+        }
+
+        return false;
+    }
+
+    function _hasStorageSlotsSet(address _addr) internal pure returns (bool) {
```

Comment:
This isn't a scalable way to write these kinds of assertions. Now a dev needs to go in and remember to update this test in the future